numpy/f2py/crackfortran.py

```diff
@@ -2433,7 +2433,7 @@ def get_parameters(vars, global_params={}):
                     v = v.replace(*repl)
             v = kind_re.sub(r'kind("\1")', v)
             v = selected_int_kind_re.sub(r'selected_int_kind(\1)', v)
-            if isinteger(vars[n]) and not selected_kind_re.match(v):
+            if not selected_kind_re.match(v):
```

Comment:
I am sorry that this commit is now much more complex... Could you please review the changes made regarding the complex case.

I realize that it still does not correctly evaluate the complex numbers. But providing a complete parser for the complex numbers is a bit difficult (and besides, all this fixes are warnings).